commit 5aa5fc887c8e2fa91f1409317d46c392718bacf0
Author: Gavin Kinsey <gavin.kinsey@hopewiser.com>
Date:   Tue Oct 11 09:46:38 2016 +0100

    Fix wxHTML parse bug in non-unicode, ? characters are not displayed.
    
    GetEntitiesParser()->GetCharForCode(NBSP_UNICODE_VALUE) in the non-unicode
    build returns the value '?' as it doesn't find a match for that value.  The
    parser then proceeds to replace all '?' characters in the HTML document with
    NBSP.
    
    Change the type of the #define to be unsigned int rather than wxChar for
    non-unicode to fix this.

diff --git a/src/html/winpars.cpp b/src/html/winpars.cpp
index edc8ffe..3fec7a5 100644
--- a/src/html/winpars.cpp
+++ b/src/html/winpars.cpp
@@ -338,10 +338,11 @@ wxFSFile *wxHtmlWinParser::OpenURL(wxHtmlURLType type,
     return GetFS()->OpenFile(myurl, flags);
 }
 
-#define NBSP_UNICODE_VALUE  (wxChar(160))
 #if !wxUSE_UNICODE
+    #define NBSP_UNICODE_VALUE  (160U)
     #define CUR_NBSP_VALUE m_nbsp
 #else
+    #define NBSP_UNICODE_VALUE  (wxChar(160))
     #define CUR_NBSP_VALUE NBSP_UNICODE_VALUE
 #endif
 
